sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala

```diff
@@ -747,6 +748,18 @@ class ExpressionTypeCheckingSuite extends SparkFunSuite with SQLHelper with Quer
     )
   }
 
+  test("hash expressions are prohibited on VariantType elements") {
+    val rawVariant = new VariantVal(Array[Byte](0, 0, 0), Array[Byte](0, 0, 0))
+    val argument = Literal.create(rawVariant, VariantType)
```

Comment:
Since the test only validates the type check phase, the actual variant value doesn't matter. You can use `Literal.create(null, VariantType)` to simplify the test a bit (no need to import `VariantVal`).

Reply:
Fixed